hazelcast-client-new/src/main/java/com/hazelcast/client/proxy/ClientReplicatedMapProxy.java

```diff
@@ -79,8 +82,14 @@
     protected static final String NULL_KEY_IS_NOT_ALLOWED = "Null key is not allowed!";
     protected static final String NULL_VALUE_IS_NOT_ALLOWED = "Null value is not allowed!";
 
+    private static final AtomicIntegerFieldUpdater<ClientReplicatedMapProxy> TARGET_PARTITION_ID_UPDATER =
+            AtomicIntegerFieldUpdater.newUpdater(ClientReplicatedMapProxy.class, "targetPartitionId");
+
     private volatile ClientHeapNearCache<Object> nearCache;
     private final AtomicBoolean nearCacheInitialized = new AtomicBoolean();
+    // all requests are forwarded to the same node via a random partition id
+    // do not use this field directly. use getOrInitTargetPartitionId() instead.
+    private volatile int targetPartitionId = Operation.GENERIC_PARTITION_ID;
```

Comment:
Sorry. I was thinking about suggesting onInitialize method rather than lazy initialization. I forget the fact that name is used in PartitionSpecificClientProxy.